pkg/forwarder/forwarder.go

```diff
@@ -91,23 +103,23 @@ type Worker struct {
 	lastMetrics []*clientmodel.MetricFamily
 	lock        sync.Mutex
 	reconfigure chan struct{}
-
-	logger log.Logger
 }
 
 // New creates a new Worker based on the provided Config. If the Config contains invalid
 // values, then an error is returned.
-func New(cfg Config) (*Worker, error) {
+func New(metrics *Metrics, cfg Config) (*Worker, error) {
```

Comment:
why do we have to do this as another argument to the `New` func and not simply create the metrics inside of this creator func? That would simplify the life of the caller.